server/src/modules/mainModule/filesFinder.ts

```diff
@@ -3,17 +3,23 @@ import path from 'path';
 import imageType from 'image-type';
 
 export function* directoryFilesGenerator(baseFolder: string, recursive: boolean = true): IterableIterator<string> {
-    const currDirFiles: string[] = fs.readdirSync(baseFolder);
-    const absoluteFiles: string[] = currDirFiles.map((file: string) => path.join(baseFolder, file))
+    let currDirFiles: string[] = [];
+    try{
+        currDirFiles = fs.readdirSync(baseFolder);
+    }
+    catch{
+        return;
```

Comment:
Are you sure this won't break the generator function?
If you return from generator it counts as .done?

Reply:
return will end recursive function execution, but not generator